scattering/scattering.py

```diff
@@ -72,29 +94,48 @@ def structure_factor(
     L = np.min(trj.unitcell_lengths)
 
     top = trj.topology
-    elements = set([a.element for a in top.atoms])
+    unique_residues = []
+
+    for a in top.residues:
```

Comment:
`unique_residues` and `residue_atoms` doesn't seem like it gets called anywhere else in the function.

Reply:
Yes, they are not being used. They will be deleted in the next commit.